src/prism.c

```diff
@@ -21711,7 +21711,7 @@ pm_strnstr(const char *big, const char *little, size_t big_length) {
  */
 static void
 pm_parser_warn_shebang_carriage_return(pm_parser_t *parser, const uint8_t *start, size_t length) {
-    if (length > 2 && start[length - 1] == '\n' && start[length - 2] == '\r') {
```

Comment:
I could have only renumbered instead:

```diff
-    if (length > 2 && start[length - 1] == '\n' && start[length - 2] == '\r') {
+    if (length > 2 && start[length] == '\n' && start[length - 1] == '\r') {
```

but IMHO it's more clear to have the `\r` and `\n` in that order, as they would be in the line ending.